packages/react-components/src/atoms/NavigationLink.tsx

```diff
@@ -79,6 +79,7 @@ const NavigationLink: React.FC<NavigationLinkProps> = ({
         activeClassName={activeClassName}
         css={[styles, { [`&.${activeClassName}`]: activePrimaryStyles }]}
         smooth
+        isActive={(match, _) => !!match}
```

Comment:
this needs a regression test

Reply:
can you please check the newly added tests?

Reply:
I had to re-do the tests unfortunately since they were relying on implementation details (a class name). I borrowed from `NavigationLink.test` the way that is used to assert on active links (checking bg color).
Note that may be of interest: The rerendering in active and non-active state to observe the difference is there to ensure the test doesn't just assert on a background color that is *always* there no matter if active or not and give a false sense of security. You'll see this pattern in many places across the component lib